src/content/docs/develop/updating-dependencies.mdx

```diff
@@ -52,6 +52,10 @@ cargo update
 
 Alternatively, you can run the `cargo upgrade` command provided by [cargo-edit] which does all of this automatically.
 
+## Sync npm Packages and Cargo Crates versions
+
+Since the JavaScript APIs rely on Rust code in the backend, adding a new feature requires upgrading both sides to ensure compatibility. Please make sure you have the same minor version of the npm package `@tauri-apps/api` and cargo crate `tauri` synced, this is also needed for the plugins, for example, you need to sync the minor version of npm package `@tauri-apps/plugin-fs` and cargo crate `tauri-plugin-fs`
```

Comment:
The plugins go a step further and are synced at patch level. We released ipc breaking changes in patch releases so syncing plugin patch versions is important.

Reply:
Do we actually break them in patch releases? I know we do synced version releases, but I always though IPC breaking changes are done in minors

Anyways, I'll open another PR to add that

Reply:
> Do we actually break them in patch releases?

yep. most of the time they come in minors but typically we have to release follow up patch releases to fix stuff that are also breaking 🙃 
either way, with the patch syncing i always thought we can do that in patches as well so at least the PRs i merged may do it in patches.

Reply:
> most of the time they come in minors

holy shit, i couldn't have phrased that worse